cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -1128,5 +1140,119 @@ static SearchTerm from(@Nullable NamespaceId namespaceId, String rawTerm) {
 
       return new SearchTerm(namespaceId, formattedTerm, isPrefix);
     }
+
+  }
+
+  /**
+   * Starts scanning of V1 table value row keys in batches. If there are no more value row keys available,
+   * it will scan history rows. If there are no value/history rows, it returns empty list.
+   *
+   * @param limit  Batch size for scan/delete.
+   * @return metadata entries with list of {@link KeyValue} where key is timestamp for history row other wise
+   * it is null, value is {@link MetadataEntry} and list of rowkeys to be deleted.
+   */
+  public MetadataEntries scanFromV1Table(int limit) {
+    MetadataEntries scannedEntries = new MetadataEntries(new LinkedList<>(), new LinkedList<>());
+
```

Comment:
It is not good to make the scan batch include value and history both.
Also how will this work. If we migrate all the value rows first and then history row and we are serving entities which are migrated so we have a situation where you can query and see metadata but there is no history for it as the history is migrated at the end.

Reply:
History rows are only used to query metadata for a program run id. If it is not migrated it can not be queried.

Reply:
I understand that. This approach is incorrect but it is too late change it. They should be migrated and made available together if you don't want that then document consequence properly.